src/core/seat/cursor.cpp

```diff
@@ -33,7 +33,10 @@ wf::cursor_t::cursor_t(wf::seat_t *seat)
     request_set_cursor.set_callback([&] (void *data)
     {
         auto ev = static_cast<wlr_seat_pointer_request_set_cursor_event*>(data);
-        set_cursor(ev, true);
+        idle_set_cursor.run_once([=] ()
+        {
+            set_cursor(ev, true);
```

Comment:
This is definitely use-after-free, the event may be freed until idle.

Reply:
Also, we have to make sure that the cursor surface doesn't get destroyed until idle